src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      const result = total;
+
+      total = 0;
+
+      return result;
+    }
+
+    total += arg;
+
+    return adder;
+  };
```

Comment:
Consider adding a check for the type of 'arg' to ensure that only numbers are added to the total. This will prevent potential bugs if the function is used with non-numeric arguments.